### ISSUE TITLE
Read configuration from trackable path

### ISSUE THREAD

**Issue by junwei-wang** [Jean-Tinland/simple-bar#145]
First of all, thank you very much for this excellent project.

I want to synchronize my configuration for simple-bar between multiple machines.
I use chezmoi (git-based) to track my configurations for other applications.
Is there a way for simple-bar to load configuration from some common configuration path, such as `.config/simple-bar/simplebarrc`.

I know I can always do soft links, but it would be nice if simple-bar support that natively.

Thanks a lot.

**Comment by Jean-Tinland** [comment]
Hi!

I was planing on adding an import/export function in settings that would create a json file on user's desktop but I can also add a setting letting the user choose an external file in which simple bar would automatically write user preferences in addition to local storage.

What do you think?

I'm glad simple-bar suits you!

**Comment by junwei-wang** [comment]
This import / export idea is excellent, but still people has to do it manually.
"The customizable user preferences path" suits me better for my purpose. 

It would be perfect for me if this customizable path by default is located at some common configuration path:
- $HOME/config/simplebar/simplebarrc
- $HOME/.simplebarrc

Thanks a lot for you reply and your passion for this project. It improves my productivity a lot ;)

**Comment by Jean-Tinland** [comment]
Hi!

I had to choose a different approach (not far from what we discussed) but I think I will suits you.

There are now 2 new buttons in the settings module: "Import" & "Export". In order to activate them (they are hidden by default) you'll need to activate the "External config file" setting:

![image](https://user-images.githubusercontent.com/43068795/123518193-19755580-d6a5-11eb-8745-d9f548339f22.png)

The "Import" button will read from "~/.simplebarrc" if existing and the "Export" one will write in this same file. **Be-careful this will overwrite the external config file each time you click on it**.

There also has been some changes made regarding the signals that needed to be set in your `.yabairc`: they are no longer required as simple-bar is now doing that itself ([see this section in the readme file](https://github.com/Jean-Tinland/simple-bar#refresh-bar-on-app-or-workspace-change)).

Also `process` and `spaces` widgets have been merged so you will need to restart both `yabai` & `Übersicht` in order to make everything work again!

Let me know this is ok for you.

And thank your for the compliment and your suggestion. I'm glad you are enjoying simple-bar! :)
